giraffe/src/components/geo/processing/PivotedGeoTable.ts

```diff
@@ -0,0 +1,94 @@
+// Libraries
+import {values} from 'lodash'
+
+// Types
+import {
+  FIELD_COLUMN,
+  filterMetaColumns,
+  GEO_HASH_COLUMN,
+  VALUE_COLUMN,
+} from './tableProcessing'
+import {AbstractGeoTable, CoordinateEncoding} from './AbstractGeoTable'
+import {Table} from '../../../types'
+
+interface GeoRow {
+  [key: string]: number | string
+}
+
+export const isPivotSensible = table => {
+  const fieldColumn = table.getColumn(FIELD_COLUMN, 'string')
+  const valueColumn = table.getColumn(VALUE_COLUMN)
+  return fieldColumn && valueColumn
+}
+
+export class PivotedGeoTable extends AbstractGeoTable {
+  data: GeoRow[]
+  truncated: boolean
+  maxRows: number
+
+  constructor(table: Table, maxRows) {
+    const seriesKeyNames = filterMetaColumns(table.columnKeys)
+    const seriesKeyColumns = seriesKeyNames.map(key => table.getColumn(key))
+    const fieldColumn = table.getColumn(FIELD_COLUMN, 'string')
+    const valueColumn = table.getColumn(VALUE_COLUMN)
+    let lonFound = false,
+      latFound = false
+    if (!fieldColumn || !valueColumn) {
+      super(CoordinateEncoding.GEO_HASH)
+      this.data = []
+      return
+    }
+    const entriesCount = fieldColumn.length
+    const index: {[seriesKey: string]: GeoRow} = {}
+    for (let i = 0; i < entriesCount; i++) {
+      const seriesKey = seriesKeyColumns.map(column => column[i]).join()
+      const fieldName = fieldColumn[i]
+      if (!lonFound && fieldName === 'lon') lonFound = true
```

Comment:
Please always use braces for if statements. We will be updating the eslint rules to reflect this.